### ISSUE TITLE
[Yaml] Yaml::dump() encodes empty array as empty object

### ISSUE THREAD

**Issue by kleijnweb** [symfony/symfony#15781]
See https://github.com/kleijnweb/swagger-bundle/issues/18

**Comment by fabpot** [comment]
Have a look at the signature and the last argument.

**Comment by kleijnweb** [comment]
I did, and I followed it to `Inline::dump()#106`, where it checks if the value is an object, which it is not. If it is, and that last argument (`$objectSupport`) is set to TRUE instead of the default FALSE, it will serialize the object and prefix it with '!!php/object:`. Clearly this is not what is happening.

Also my assertion `$this->assertNotRegExp('/\: \{  \}/', $content);` will still fail, even with the argument set to FALSE (`$yaml = Yaml::dump($data, 10, 2, false, true);`).

I am starting to doubt myself a little bit here, but right now I think this should be reopened.

**Comment by xabbuh** [comment]
@kleijnweb see also #9870

**Comment by xabbuh** [comment]
see #17728 for the implementation

**Comment by didoda** [comment]
```
$yaml = ['test' => []];
$result = Yaml::dump($yaml, 2, 4, Yaml::DUMP_OBJECT_AS_MAP);
echo $result; // => "test: {  }"
```

it seems to me that "empty array" return "empty object"; how can I get "test: [ ]" as result (empty array)?

**Comment by xabbuh** [comment]
@didoda You need to use the `DUMP_EMPTY_ARRAY_AS_SEQUENCE` flag which will be available in Symfony 3.3 (see #21471).

**Comment by didoda** [comment]
@xabbuh , thanks for the reply. I will use `DUMP_EMPTY_ARRAY_AS_SEQUENCE` when available ;)
